packages/react-core/src/components/Divider/__tests__/Divider.test.tsx

```diff
@@ -1,34 +1,98 @@
+import { render, screen } from '@testing-library/react';
+import { Divider } from '../Divider';
 
-test('divider using hr', () => {
-  const { asFragment } = render(<Divider />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with only the class pf-v5-c-divider by default', () => {
+  render(<Divider data-testid="divider" />);
+  expect(screen.getByTestId('divider')).toHaveClass('pf-v5-c-divider', { exact: true });
 });
 
-test('divider using li', () => {
-  const { asFragment } = render(<Divider component="li" />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with horizontal rule by default', () => {
+  render(<Divider />);
+  expect(screen.getByRole('separator')).toContainHTML('<hr class="pf-v5-c-divider" />');
 });
 
-test('divider using div', () => {
-  const { asFragment } = render(<Divider component="div" />);
-  expect(asFragment()).toMatchSnapshot();
+test('Renders with vertical divider', () => {
+  render(<Divider orientation={{ default: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical');
+});
+
+test('Renders with li', () => {
+  render(<Divider component="li" />);
+  expect(screen.getByRole('separator')).toContainHTML('<li class="pf-v5-c-divider" role="separator" />');
+});
+
+test('Renders with div', () => {
+  render(<Divider component="div" />);
+  expect(screen.getByRole('separator')).toContainHTML('<div class="pf-v5-c-divider" role="separator" />');
 });
 
-test('vertical divider', () => {
-  const { asFragment } = render(
-    <Flex>
-      <FlexItem>first item</FlexItem>
-      <Divider
-        orientation={{
-          default: 'vertical'
-        }}
-      />
-      <FlexItem>second item</FlexItem>
-    </Flex>
-  );
+test('Test default orientation', () => {
+  render(<Divider orientation={{ default: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical');
+});
+
+test('Test sm orientation', () => {
+  render(<Divider orientation={{ sm: 'horizontal' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-horizontal-on-sm');
+});
+
+test('Test md orientation', () => {
+  render(<Divider orientation={{ md: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical-on-md');
+});
+
+test('Test lg orientation', () => {
+  render(<Divider orientation={{ lg: 'horizontal' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-horizontal-on-lg');
+});
+
+test('Test xl orientation', () => {
+  render(<Divider orientation={{ xl: 'vertical' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-vertical-on-xl');
+});
+
+test('Test 2xl orientation', () => {
+  render(<Divider orientation={{ '2xl': 'horizontal' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-horizontal-on-2xl');
+});
+
+test('Test default inset', () => {
+  render(<Divider inset={{ default: 'insetNone' }} />);
+  expect(screen.getByRole('separator')).toHaveClass('pf-m-inset-none');
+});
```

Comment:
For these inset tests, we should try to test that the applicable class is applied for each combination of each key and value. So we'd want to make sure `pf-m-inset-sm-on-sm` applies and also `pf-m-inset-3xl-on-sm` applies, etc.

Reply:
i added some code that should test all of the possible insets, should do the trick